src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1169,11 +1175,17 @@ def set_up_unbind_service_bindings(self):
                 new_template["serviceBinds"] = existing_template.get("serviceBinds", [])
 
             service_bindings_dict = {}
+            java_component_name_set = {}
             if new_template["serviceBinds"]:
                 service_bindings_dict = {service_binding["name"]: index for index, service_binding in
                                          enumerate(new_template.get("serviceBinds", []))}
+                java_component_name_set = {binding["name"].replace('_', '-') for binding in new_template["serviceBinds"]
+                                           if is_valid_java_component_resource_id(binding["serviceId"])}
 
             for item in self.get_argument_unbind_service_bindings():
+                if item in java_component_name_set and '-' in item:
```

Comment:
ni:
Please add comment about if resource is java component type, we will auto change the `-` in binding name to `_`.

Also, should add info message to customer about it:
`logger.info("auto change the "-" in binding name to "_".")`

Reply:
fixed, thanks a lot